mcs/class/System/System.Diagnostics/PerformanceCounter.cs

```diff
@@ -138,6 +141,16 @@ public PerformanceCounter ()
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
 		static extern void FreeData (IntPtr impl);
 
+		static int StringLength (string a)
```

Comment:
All the `categoryName`, `counterName`, `machineName` and `instanceName` are already guaranteed to be non-`null`, so all these null checks are redundant. All the calls to `StringLength(categoryName)` can safely be replaced by `categoryName.Length` for example.

Reply:
I didn't realize that.